src/model/actor.ts

```diff
@@ -0,0 +1,5 @@
+export interface Actor {
```

Comment:
Have you checked any existing packages on npm: https://www.npmjs.com/search?q=activitypub 

Maybe they make a lot of boiler plating for you obsolete.